lib/mongoid/traversable.rb

```diff
@@ -27,7 +27,7 @@ def _parent=(p)
     #
     # @return [ Array<Document> ] All child documents in the hierarchy.
     def _children
-      @__children ||= collect_children
```

Comment:
The performance improvement is not worth the danger of the example in #3991. If you think the memoization is highly important, I'd recommend merging this, releasing the bugfix, and then adding memoization back in an improved way as a performance improvement dot release. As it is now, silently failing is terrible.